### PR TITLE
Revert "quick fix for CircleCI (#2279)"

### PR DESCRIPTION
**Pull request by melekes**
This reverts commit 1cf6712a36e8ecc843a68aa373748e89e0afecba.

Plus, switch from gometalinter to golangci-lint https://github.com/golangci/golangci-lint for speed and better support.

### CODE REVIEW

**Comment by Liamsi** [comment]
ref: https://github.com/tendermint/tendermint/pull/2279
background: saving cache for PR from forks was broken on circleci
Is this fixed now?

**Comment by melekes** [comment]
> Is this fixed now?

looks like it

File: .golangci.yml [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    - dupl
+    - ineffassign
+    - interfacer
+    - unconvert
+    - goconst
+    - unparam
+    - nakedret
+    - lll
+    - gochecknoglobals
+    - govet
```
Comment (Liamsi) [inline .golangci.yml]: We do not pass govet? Wasn't this enabled in metalinter? 
https://github.com/tendermint/tendermint/pull/3261/commits/da33dd04cc90f60f339afa8182f59294e67d151b#diff-b67911656ef5d18c4ae36cb6741b7965L252

File: .golangci.yml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    - gocritic
+    - gosec
```
Comment (Liamsi) [inline .golangci.yml]: same, wasn't this previously enabled?

File: .golangci.yml [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    - gochecknoinits
+    - scopelint
+    - stylecheck
+    - deadcode
```
Comment (Liamsi) [inline .golangci.yml]: same:
https://github.com/tendermint/tendermint/pull/3261/commits/da33dd04cc90f60f339afa8182f59294e67d151b#diff-b67911656ef5d18c4ae36cb6741b7965L254

File: .golangci.yml (hunk 1 above)
Comment (melekes) [reply .golangci.yml]: probably because of the older version

Comment (Liamsi) [reply .golangci.yml]: Weird, I thought govet is just part of the golang distribution (the `go vet` command).

**Comment by Liamsi** [comment]
I think, we should merge this as is and do a follow-up PR which re-enables all previously running linters (vet, gosec, ...?) and do the changes that are necessary due to the newer versions of these linters (and decide / discuss if want to enable a few more).

File: .golangci.yml (hunk 1 above)
Comment (melekes) [reply .golangci.yml]: Or maybe gometalinter was broken

**Comment by melekes** [comment]
Yep, created https://github.com/tendermint/tendermint/issues/3262

File: blockchain/pool.go [hunk 4]
```diff
@@ -363,23 +363,23 @@ func (pool *BlockPool) sendError(err error, peerID p2p.ID) {
...
-			str += fmt.Sprintf("H(%v):X ", h)
-		} else {
-			str += fmt.Sprintf("H(%v):", h)
-			str += fmt.Sprintf("B?(%v) ", pool.requesters[h].block != nil)
-		}
-	}
-	return str
-}
+// for debugging purposes
+// func (pool *BlockPool) debug() string {
```
Comment (Liamsi) [inline blockchain/pool.go]: I wish there was a way to tell the linter: ignore this function (like for instance with clippy in rust).

Comment (Liamsi) [reply blockchain/pool.go]: Wait, doesn't `nolint:` work for golangci, too?

Comment (melekes) [reply blockchain/pool.go]: It does. Do you think we should add such labels instead of commenting?

Comment (Liamsi) [reply blockchain/pool.go]: Yes, (in the case of deadcode) if we consider the code to be useful, I would prefer a nolint there. If it's clear that it isn't useful and won't be used (for debugging or alike), I would prefer to delete it (even better).

Comment (melekes) [reply blockchain/pool.go]: Done

Comment (Liamsi) [reply blockchain/pool.go]: Awesome! This also makes it easier to search for these places.

**Pull request merged**